app/controllers/identities_controller.rb

```diff
@@ -8,44 +8,56 @@ class IdentitiesController < ApplicationController
 
   def update
     handle_with(IdentitiesUpdate,
-                success: lambda { render status: :accepted, text: 'Password changed' },
-                failure: lambda {
+                success: lambda  do
+                  security_log :password_updated
+                  render status: :accepted, text: 'Password changed'
+                end,
+                failure: lambda do
                   render status: 400, text: @handler_result.errors.map(&:message).to_sentence
-                })
+                end)
   end
 
   def reset_password
     if !current_user.is_anonymous?
       if current_user.identity.nil?
+        security_log :password_reset_failed
         flash[:alert] = "You cannot reset your password because your account does not have a password."
         redirect_to profile_path
         return
       end
 
       if current_user.identity.password_expired?
+        security_log :password_expired
         store_fallback key: :password_return_to
         flash[:alert] = 'Your password has expired. Please enter a new password.'
       end
     end
+
     handle_with(IdentitiesResetPassword,
-                success: lambda {
+                success: lambda do
                   return if !request.post?
                   sign_in! @handler_result.outputs[:identity].user
+                  security_log :password_reset
                   redirect_back key: :password_return_to,
                                 notice: 'Your password has been reset successfully! You are now signed in.'
-                },
-                failure: lambda {
+                end,
+                failure: lambda do
+                  security_log :password_reset_failed
                   render :reset_password, status: 400
-                })
+                end)
   end
 
 
   def destroy
     handle_with(AuthenticationDelete,
-                success: lambda { render status: :ok, text: "#{params[:provider].titleize} removed" },
-                failure: lambda {
+                success: lambda do
+                  security_log :authentication_deleted,
+                               authentication_id: @handler_result.outputs.authentication.id
```

Comment:
can we add the provider info to created and deleted?  Possible that we don't need it in deleted if have it in created and can look at both log records.